src/Toolkit/Toolkit.Maui/ContentControl.cs

```diff
@@ -0,0 +1,81 @@
+﻿#if MAUI
+using Microsoft.Maui.Controls.Internals;
+
+namespace Esri.ArcGISRuntime.Toolkit.Maui
+{
+    internal class ContentControl : TemplatedView
+    {
+        private static readonly ControlTemplate DefaultControlTemplate;
+
+        static ContentControl()
+        {
+            DefaultControlTemplate = new ControlTemplate(BuildDefaultTemplate);
+        }
+
+        private static object BuildDefaultTemplate()
+        {
+            var presenter = new ContentPresenter();
+            INameScope nameScope = new NameScope();
+            NameScope.SetNameScope(presenter, nameScope);
+            nameScope.RegisterName("ContentPresenter", presenter);
+            return presenter;
+        }
+
+        public ContentControl()
+        {
+            ControlTemplate = DefaultControlTemplate;
+        }
+
+        protected override void OnApplyTemplate()
+        {
+            UpdateData();
+            base.OnApplyTemplate();
```

Comment:
Should we let base OnApplyTemplate run first, before our update code?  That's standard practice in WPF and WinUI, and I believe MAUI follows the same rule.